### PR TITLE
[`Docs`] Fix typos, update, and add visualizations at Using Diffusers' Training page

### PR DESCRIPTION
**Pull request by standardAI**
# What does this PR do?

https://huggingface.co/docs/diffusers/main/en/training/overview:
Tried to solve linking problem, but not sure if solved -> *: Supports \[Stable Diffusion XL](../api/pipelines/stable_diffusion/stable_diffusion_xl).

https://huggingface.co/docs/diffusers/main/en/training/dreambooth#if-stage-ii-full-dreambooth:
Chose --learning_rate=5e-6 but talking about choosing a small learning rate such as 1e-8.

https://huggingface.co/docs/diffusers/main/en/training/instructpix2pix:
The generated image is a result of "make the mountains snowy", not "turn the clouds rainy". I guess the generated image should have been this -> https://twitter.com/RisingSayak/status/1628392222419992577.

https://huggingface.co/docs/diffusers/main/en/training/custom_diffusion:
model_id = "sayakpaul/custom-diffusion-cat" and model_id = "sayakpaul/custom-diffusion-cat-wooden-pot" give an error:
RuntimeError: Error(s) in loading state_dict for CustomDiffusionAttnProcessor:
	Unexpected key(s) in state_dict: "to_k_custom_diffusion.bias", "to_v_custom_diffusion.bias". 

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [X] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu @yiyixuxu @DN6

### CODE REVIEW

File: docs/source/en/training/adapt_a_model.md [hunk 1]
```diff
@@ -6,24 +18,24 @@ This guide will show you how to adapt a pretrained text-to-image model for inpai
...
-A [`UNet2DConditionModel`] by default accepts 4 channels in the [input sample](https://huggingface.co/docs/diffusers/v0.16.0/en/api/models#diffusers.UNet2DConditionModel.in_channels). For example, load a pretrained text-to-image model like [`runwayml/stable-diffusion-v1-5`](https://huggingface.co/runwayml/stable-diffusion-v1-5) and take a look at the number of `in_channels`:
+A [`UNet2DConditionModel`] by default accepts 4 channels in the [input sample](https://huggingface.co/docs/diffusers/main/en/api/models/unet2d-cond#diffusers.UNet2DConditionModel.in_channels). For example, load a pretrained text-to-image model like [`runwayml/stable-diffusion-v1-5`](https://huggingface.co/runwayml/stable-diffusion-v1-5) and take a look at the number of `in_channels`:
 
 ```py
 from diffusers import StableDiffusionPipeline
 
 pipeline = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5", use_safetensors=True)
 pipeline.unet.config["in_channels"]
-4
+# 4
```
Comment (stevhliu) [inline docs/source/en/training/adapt_a_model.md:28]:
```suggestion
"4"
```

File: docs/source/en/training/adapt_a_model.md [hunk 2]
```diff
@@ -6,24 +18,24 @@ This guide will show you how to adapt a pretrained text-to-image model for inpai
...
 Inpainting requires 9 channels in the input sample. You can check this value in a pretrained inpainting model like [`runwayml/stable-diffusion-inpainting`](https://huggingface.co/runwayml/stable-diffusion-inpainting):
 
 ```py
 from diffusers import StableDiffusionPipeline
 
-pipeline = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-inpainting", use_safetensors=True)
+pipeline = StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-inpainting")
 pipeline.unet.config["in_channels"]
-9
+# 9
```
Comment (stevhliu) [inline docs/source/en/training/adapt_a_model.md:38]:
```suggestion
"9"
```

**Review by stevhliu — COMMENTED** [review]
Thanks for working on the training docs! I'm actually in the middle of updating the training docs (see #5512), so let's hold off on the other pages for now except for `adapt_a_model.md` and `create_dataset.md`.

File: docs/source/en/training/adapt_a_model.md (hunk 1 above)
Comment (standardAI) [reply docs/source/en/training/adapt_a_model.md:28]: If we do this, the cell prints "4" not the result of pipeline.unet.config["in_channels"].

File: docs/source/en/training/adapt_a_model.md (hunk 2 above)
Comment (standardAI) [reply docs/source/en/training/adapt_a_model.md:38]: If we do this, the cell prints "9" not the result of pipeline.unet.config["in_channels"].

**Comment by standardAI** [comment]
> Thanks for working on the training docs! I'm actually in the middle of updating the training docs (see #5512), so let's hold off on the other pages for now except for `adapt_a_model.md` and `create_dataset.md`.

Hmm OK. I should have seen opened PRs.

**Pull request closed without merging**